R/call_upload_user_files.R

```diff
@@ -70,23 +131,31 @@ call_upload_user_files <- function(
   resource_filters = NA,
   resource_weight = NA,
   geo = "city",
-  acs_data_year = "2021"
-  ) {
-
-  #Define api URL
-  api_url <- stringr::str_glue("{base_url}/api/v1/upload-user-file/")
-
-  #Data Input Type Checks:
-  stopifnot(is.character(c(resource_file_path,
-                           resource_lat_column,
-                           resource_lon_column,
-                           geo,
-                           acs_data_year)
-                         )
+  acs_data_year = getOption("sedtR.year", 2021),
+  placement = "surface",
+  ...,
+  call = caller_env()
+) {
+  # Define api URL
+  api_url <- sedt_url("upload-user-file/")
+
+  resource_file_path <- resource_file_path %||% prep_sedt_resource(
+    resource = resource,
+    coords = c(resource_lon_column, resource_lat_column),
+    placement = placement,
+    ...,
+    call = call
+  )
+
+  check_resource_file_path(
+    resource_file_path,
+    coords = c(resource_lon_column, resource_lat_column),
+    call = call
   )
 
-  stopifnot(acs_data_year %in%  c("2019", "2021", "2022"))
-  stopifnot(geo %in% c("city", "county", "state", "national"))
```

Comment:
It looks like the `match_acs_data_year()` and `match_geo()` functions perform the same check of the `acs_data_year` and `geo` arguments for valid values using `arg_match0` instead of `stopifnot`. I wasn't totally clear on why this is a preferable approach?

Reply:
`rlang::arg_match0()` gives you a much more information error message. For example, if you pass 2018 (or "2018") for validation. This is what you get with arg_match (or `arg_match0()`):

Error:
! `var` must be one of "2019", "2021", or "2022", not
  "2018".
ℹ Did you mean "2019"?

`stopifnot()` gives you:

Error: var %in% as.character(c(2019, 2021, 2022)) is not TRUE

Moving validation into a helper function also reduces the number of places you'd need to change the vector of supported years if you validating an input year in more than one location.